statistics/index.go

```diff
@@ -127,23 +127,30 @@ func (idx *Index) TotalRowCount() float64 {
 	return idx.Histogram.TotalRowCount()
 }
 
-// IsInvalid checks if this index is invalid.
-func (idx *Index) IsInvalid(sctx sessionctx.Context, collPseudo bool) (res bool) {
-	idx.CheckStats()
+// IndexStatsIsInvalid checks whether the index has valid stats or not.
+func IndexStatsIsInvalid(idxStats *Index, sctx sessionctx.Context, coll *HistColl, cid int64) (res bool) {
```

Comment:
The first argument `idxStats` seems can be removed? Since `idxStats = coll.Index[cid]`.

Reply:
Here just want to save once of the accessing the map(

Reply:
`Once of accessing the map` seems OK? 
If we keep both `*Index` and `*HistColl` here, it seems unsafe since there is an implicit assumption that `the *Index must come from the *HistColl`. 
Other developers may not know this implicit assumption, and what if the `*Index` is not from `*HistColl`...